src/RelativeStream.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @see       http://github.com/zendframework/zend-diactoros for the canonical source repository
+ * @copyright Copyright (c) 2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-diactoros/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Diactoros;
+
+use Psr\Http\Message\StreamInterface;
+
+class RelativeStream implements StreamInterface
+{
+    /**
+     * @var StreamInterface
+     */
+    private $decoratedStream;
+
+    /**
+     * @var int
+     */
+    private $offset;
+
+    /**
+     * Class constructor
+     *
+     * @param StreamInterface $decodatedStream
+     * @param int $offset
+     */
+    public function __construct(StreamInterface $decodatedStream, $offset)
+    {
+        $this->decoratedStream = $decodatedStream;
+        $this->offset = $offset;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        $this->seek(0);
+        return $this->getContents();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close()
+    {
+        $this->decoratedStream->close();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function detach()
+    {
+        return $this->decoratedStream->detach();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSize()
+    {
+        return $this->decoratedStream->getSize() - $this->offset;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function tell()
+    {
+        return $this->decoratedStream->tell() - $this->offset;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function eof()
+    {
+        return $this->decoratedStream->eof();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSeekable()
+    {
+        return $this->decoratedStream->isSeekable();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function seek($offset, $whence = SEEK_SET)
+    {
+        switch ($whence) {
```

Comment:
Avoid using the `switch` here. An `if` and early return is sufficient and simpler to read.